src/briefcase/platforms/linux/flatpak.py

```diff
@@ -1 +1,232 @@
 # An implementation would go here!
+
+# needs system tools: flatpak and flatpak-builder
+#
+# also I used https://github.com/flatpak/flatpak-builder-tools/blob/master/pip/flatpak-pip-generator
+# to transform eg: 'requirements.txt' into the "python3-requirements.json"
+# 
+# but really we want to transform the pyproject.toml directly and to have the
+# transformer be a library not a standalone python program.
+# 
+# commands run:
+#
+# flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+# flatpak install flathub org.freedesktop.Platform//21.08 org.freedesktop.Sdk//21.08
+# pip freeze > requirements.txt
+# flatpak-pip-generator --requirements-file=requirements.txt
+#
+# org.zoic.hello.yaml:
+#
+#   app-id: org.zoic.hello
+#   runtime: org.freedesktop.Platform
+#   runtime-version: '21.08'
+#   sdk: org.freedesktop.Sdk
+#   command: hello.py
+#   modules:
+#     - name: hello
+#       buildsystem: simple
+#       build-commands:
+#         - install -D hello.py /app/bin/hello.py
+#       sources:
+#         - type: file
+#           path: hello.py
+#     - python3-requirements.json
+#   finish-args:
+#     - "--socket=fallback-x11"
+#     - "--socket=wayland"
+#
+# flatpak-builder --user --install --force-clean build-dir/ org.zoic.hello.yaml
+# flatpak run -v org.zoic.hello
+#
+# and that got the app to pop up on my machine ...
+#
+# Issues:
+# * the flakpak-builder manifest file is either json or yaml, it'll parse JSON
+#   from a tempfile or /dev/stdin but it interprets include paths relative so
+#   use absolute paths if that's what you want to do.
+# * flatpak-builder is in C (!)
+# * flatpak-pip-generator is a python program which translates requirements.txt
+#   files into includable "python3-requirements.json" module *but* it isn't 
+#   distributed on pypi, isn't a library and doesn't load pyproject.toml so I
+#   suspect we're better off doing this stuff here instead.
```

Comment:
What Python version is being used here? One of the implied contracts of Briefcase packaging is that it uses the same Python version as the one doing the packaging. If I package this app on Ubuntu 22.04 (which uses Python 3.9 as a system Python IIRC), and then deploy on Fedora 28 which uses Python 3.6... what happens? What happens if my code uses the walrus operator? I'm not seeing anything in this flatpak definition that references a Python version.

Reply:
Yep, the python version comes from the flatpak "runtime", which is kind of a base image for the flatpak, itself also a flatpak.

The runtime is specified in the flatpak manifest https://github.com/beeware/briefcase/pull/754/files#diff-c3953738384a9db4f1f0e7c5000448f194a3906c3260fd510190dfb02a05d887R131-R136 ...
I'm just using an arbitrary one, but I'd considered using a --option to let the user choose one.

We could also pick the appropriate runtime to match the python version being used for packaging, plus it doesn't look too difficult to create custom runtimes based on the existing ones so we could make & publish our own ones per python version (though I haven't tried this yet)

Reply:
What's the line between "tool provided by the Runtime" and "binary dependency required to make the app run"?

In particular - how would the introduction of something like #628 interact with the choice of runtime?